controllers/common.go

```diff
@@ -44,3 +47,18 @@ func getAtlasClientFromSecret(secret *corev1.Secret) (*mongodbatlas.Client, stri
 
 	return client, atlasGroupID, nil
 }
+
+func getClusterNameFromHostTemplate(ctx context.Context, client *mongodbatlas.Client, groupID, hostTemplate string) (string, error) {
+	clusters, _, err := client.Clusters.List(ctx, groupID, &mongodbatlas.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	for _, cluster := range clusters {
+		if strings.Contains(cluster.SrvAddress, hostTemplate) {
```

Comment:
can we use instead url.Parse() then Hostname() instead? this seems like a bug *can* occur if subdomain difference like smaller.